pkg/reconciler/route/route.go

```diff
@@ -172,6 +172,7 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	// assumptions about defaulting.
 	r.SetDefaults(v1beta1.WithUpgradeViaDefaulting(ctx))
 	r.Status.InitializeConditions()
+	r.Status.ObservedGeneration = r.Generation
```

Comment:
Okay so I spent some more time looking at this. As not all returned errors from this method result in Conditions being updated to "False" which was my original assumption. We likely have some other bugs to fix in this reconciler because of this, but to prevent scope creep we should move it back to where you originally had it.

So this should be on:
1. Line 229 after `configureTraffic` fails (where you originally had it)
2. Line 287 after we reconcile successfully

Sorry for the back and forth here.